src/js/IAMClient.js

```diff
@@ -1,11 +1,24 @@
 // @noflow
 
-import type { AppState } from '../types/state';
+import type { AppState, AssumeRoleParams, OIDCState } from '../types/state';
 import type { Credentials } from '../types/zenko';
 import IAM from 'aws-sdk/clients/iam';
 import type { IAMClient as IAMClientInterface } from '../types/iam';
 import { getClients } from '../react/utils/actions';
 
+export const roleName = 'scality-internal/storage-manager-role';
+
+export function getAssumeRoleWithWebIdentityParams(
+  oidc: OIDCState,
+  accountID: string,
+): AssumeRoleParams {
+  return {
+    idToken: oidc.user.id_token,
+    roleArn: `arn:aws:iam::${accountID}:role/${roleName}`,
+    RoleSessionName: oidc.user.profile.sub,
```

Comment:
I understand the GDPR concerns, but it was decided to use email for auditing purposes and identifying the user.
Would you be able to create a documentation ticket on how we can find identity of a user from the Sub(session name)?
You can link it as a blocker for https://scality.atlassian.net/browse/ARTESCA-2553

Reply:
The sub claim is the user unique identifier `<user_id>` from keycloak. To get the user information from this ID, we can either go to 
`<keycloak_url>/auth/admin/master/console/#/realms/<realm_name>/users/<user_id>` or, if we want to get it programmatically, we can use keycloak's API: [https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_getuser](https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_getuser).

The identity is indeed an email as per the design, but there is no clear mapping between the use of this email in the logs and how to identify the session initiator: `To assist in tracing the 'origin' of such requests, the RoleSessionName is provided to identify the particular assumption.`. The user id is opaque and can (should) be used to identify the user while complying with GDPR.

Example:
```
eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJZNUU4LW52b1RZeE43ekpEVktOT1lKWnFzd2dVSWZJOFJLOXJTTExZQTNvIn0.eyJleHAiOjE2MzgxNzE3NjIsImlhdCI6MTYzODE3MTQ2MiwianRpIjoiYmMyZGQzZGItYzE1YS00N2YxLThhZDYtZDVkNGM3NmIyNTc5IiwiaXNzIjoiaHR0cHM6Ly8xMjcuMC4wLjE6ODQ0My9hdXRoL3JlYWxtcy9teXJlYWxtIiwiYXVkIjoiYWNjb3VudCIsInN1YiI6ImE1YTc0NTUyLTJjMWYtNDk5ZC1iZmNkLWYzN2FlMDBiMTM1YyIsInR5cCI6IkJlYXJlciIsImF6cCI6Im15Y2xpZW50Iiwic2Vzc2lvbl9zdGF0ZSI6IjI2MWQ2MjhiLTcxOGMtNDZiZC05M2NkLTgzMWU2NTQ5ZjczNSIsImFjciI6IjEiLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJTdG9yYWdlTWFuYWdlciIsInVtYV9hdXRob3JpemF0aW9uIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJwcm9maWxlIGVtYWlsIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInJvbGVzIjpbIm9mZmxpbmVfYWNjZXNzIiwiU3RvcmFnZU1hbmFnZXIiLCJ1bWFfYXV0aG9yaXphdGlvbiJdLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJjYWU5MWU0NWFlZDgwZjNhM2ZlMjg1YzNjOGMxYTdlNzhkODJkNDczIiwiZW1haWwiOiJjYWU5MWU0NWFlZDgwZjNhM2ZlMjg1YzNjOGMxYTdlNzhkODJkNDczQHRlc3QuY29tIiwiZ3JvdXAiOltdfQ.utE8nC1nKg6g3NM5GxnmuAGulWtiSwHYNcMoirzxx_UAkYUbrkCI-azKJfsEjF_fkp4g1ACWsNiMUjtiVmq_kO-qLqAcvk0DYZmvNMEUV6u28SVHvSA_2N-6xFI4Oxo9hhEZ9FSc-WWji3qJVXuB3P3o1gR36I-qnIhLo6RLap8e-S90khBAMX2lnNuDpsfGeEYET_dMWJ2ZvzikwdJK5UNhAYl4RUo-kRcataoCxcSHbAmMebaVCn52VHMl5j-PCV2gZt6KXL_gF-k4vI1vsX0_lVu2z8q-WyyzdD-J_w6C7qXH8Hqb9W-VVHJMBUmAHobFwspjrAVh3nZjia1_mA
```

Wille have the following sub claim: `"sub": "a5a74552-2c1f-499d-bfcd-f37ae00b135c",`

Later on, we can find the user in keycloak using this id:
![image](https://user-images.githubusercontent.com/91462779/143827826-4ea690f1-191c-466d-87f5-4172f031fa29.png)

Reply:
Ticket added.

Reply:
By the way, it is still possible for the administrator, if he/she wants to, to set the `sub` claim value to be the user's email.

Reply:
cool, I agree with your approach, just making sure we document that.

Reply:
It looks like the role session name includes the service name in Pensieve Api. Should we follow the same logic for zenko-ui project? https://github.com/scality/pensieve-api/pull/143/files#diff-fa3ccffbf25e42d6d0eeba977259607da1812212f3df390d2c6a0523e8018bffR63

Reply:
For me the idea was to prefix the service name when it wasn't ZKUI, but I don't see any problem with adding zenko-ui here as well. If you think it's a good thing to do, let me know, I'll reopen a PR with this change!